### PR TITLE
Sync issues with news

### LINKED ISSUE #338: Synchronize GitHub issue labels with news file types
It would be easier to comprehend if there were labels connected to news file types. The documentation will then contain this description of news file `{issue-id}.{issue-label}`, which will be much more easier to understand than the current description.

### PR DESCRIPTION
**Pull request by ozkanonur**
Fixes #338

### CODE REVIEW

**Comment by ozkanonur** [comment]
I tried to simplify docs but for merging this, we need to provide new labels (``type:bug``, ``type:feature``, etc) and attach them to current issues (or pull requests that are not related to issues)

**Comment by Zlopez** [comment]
Something must be wrong, I see 40 commits in this PR. Could you rebase it to latest master?

**Comment by ozkanonur** [comment]
> Something must be wrong, I see 40 commits in this PR. Could you rebase it to latest master?

Sorry, I didnt notice that. Should be fine now.

File: docs/dev-guide.rst [hunk 1]
```diff
@@ -206,26 +206,15 @@ Release notes
 =============
 
 To add entries to the release notes, create a file in the ``news`` directory
-with the ``source.type`` name format, where ``type`` is one of:
-
-* ``feature``: for new features
```
Comment (Zlopez) [inline docs/dev-guide.rst:211]: I don't think it's good idea to remove this text. We should still have list of supported labels.

File: docs/dev-guide.rst [hunk 2]
```diff
@@ -206,26 +206,15 @@ Release notes
...
-* ``bug``: for bug fixes
-* ``api``: for API changes
-* ``dev``: for development-related changes
-* ``author``: for contributor names
-* ``other``: for other changes
-
-And where the ``source`` part of the filename is:
```
Comment (Zlopez) [inline docs/dev-guide.rst:218]: Same as before, this still needs to be described here.

File: docs/dev-guide.rst [hunk 3]
```diff
@@ -206,26 +206,15 @@ Release notes
...
-
-* ``42`` when the change is described in issue ``42``
-* ``PR42`` when the change has been implemented in pull request ``42``, and
-  there is no associated issue
-* ``username`` for contributors (``author`` extention). It should be the
-  username part of their commit's email address.
-  
+with the ``id.type`` name format, where ``type`` is either label of the
```
Comment (Zlopez) [inline docs/dev-guide.rst:209]: It's not clear from this descriptiong what label should be used. I would use something like `type is label of the issue that is named label.type (see supported types above)`. If the issue or PR is missing a label, please ask maintainer to add one.

File: docs/dev-guide.rst [hunk 4]
```diff
@@ -206,26 +206,15 @@ Release notes
...
+issue or the pull request.
+
+And where the ``id`` part of the filename is either id of the issue or the pull
```
Comment (Zlopez) [inline docs/dev-guide.rst:212]: This is described clearly above.

File: news/338.dev [hunk 5]
```diff
@@ -0,0 +1 @@
+Update filename format of news to sync issues with news
```
Comment (Zlopez) [inline news/338.dev:1]:
```suggestion
Update filename format of news file to correspond with labels for issues
```

**Review by Zlopez — CHANGES_REQUESTED** [review]
I don't think we should remove the whole section that was there before, this will still be true, we just add the label so it's more clear to know what type to use.

One more thing, could you switch `source` and `type` explanation in the text, so it's in the same order as in the name of the file. `source` explanation first, `type` second.

File: docs/dev-guide.rst (hunk 3 above)
Comment (ozkanonur) [reply docs/dev-guide.rst:209]: I thought `label` points to the `type`. Shouldn't be something like `id.label` instead of `label.type`?

Comment (Zlopez) [reply docs/dev-guide.rst:209]: I wrote it backwards, it should be the name of the label `type.label`, so it's easy to recognize among other labels.

File: docs/dev-guide.rst [hunk 6]
```diff
@@ -206,27 +206,28 @@ Release notes
...
-* ``PR42`` when the change has been implemented in pull request ``42``, and
-  there is no associated issue
-* ``username`` for contributors (``author`` extention). It should be the
-  username part of their commit's email address.
   
 For example:
 
-If this PR is solving bug 714 the file inside ``news`` should be called ``714.bug``
-and the content of the file would be:
+If this PR is solving issue #714 with labeled as ``type.bug``, the file inside news should be called 714.bug (PR714.bug if the PR does not related with any issue) and the content of the file would be:
```
Comment (Zlopez) [inline docs/dev-guide.rst:230]:
```suggestion
If this PR is solving issue #714 labeled as ``type.bug`` and named "Javascript error on add project page", the file inside news should be called 714.bug (PR714.bug if the PR does not have any linked issue and the PR number is 714) and the content of the file would be:
```

**Review by Zlopez — CHANGES_REQUESTED** [review]
Just some wording.

**Comment by ozkanonur** [comment]
> Just some wording.

Sorry, I shouldn't open this PR probably since my grammar is not perfect :D

**Review by Zlopez — APPROVED** [review]
Now I just need to create the labels :-D

**Comment by Zlopez** [comment]
The labels are now available, merging this PR.

**Pull request merged**